modules/scala-support/src/main/scala/dev/guardrail/generators/scala/circe/CirceRefinedProtocolGenerator.scala

```diff
@@ -27,11 +28,13 @@ object CirceRefinedProtocolGenerator {
     import scala.meta._
     tpe match {
       case t"String" =>
-        val pattern = prop.downField("pattern", _.getPattern).unwrapTracker
-        Target.pure(pattern.fold(tpe) { pat =>
-          val refined = s"""_root_.eu.timepit.refined.string.MatchesRegex[$className.`"$pat"`.T]""".parse[Type].get
-          t"""String Refined $refined"""
-        })
+        prop.downField("pattern", _.getPattern).fold(Target.pure(tpe)) { patternTracker =>
+          Try {
+            val pat = patternTracker.unwrapTracker
+            val refined = s"""_root_.eu.timepit.refined.string.MatchesRegex[$className.`"$pat"`.T]""".parse[Type].get
+            t"""String Refined $refined"""
+          }.fold(th => Target.raiseUserError(s"$th: ${patternTracker.showHistory}"), Target.pure)
```

Comment:
should it be Target.raiseException?

Reply:
Fortunately this went away when `.parse` was removed -- Perhaps there should have been a better combinator, but the point you raise is a good one -- when to use it and how to format the error message